### PR TITLE
[12.0] [REF] Account Payment Order: split generate_move method

### PR DESCRIPTION
**Pull request by ThomasBinsfeld**
Split the heavy `generate_move` for easier overrides.

### CODE REVIEW

**Comment by sbidoul** [comment]
@ThomasBinsfeld travis was red for Odoo but green for OCB. I restarted the job.

**Review by sbidoul — APPROVED** [review]
It's green now.

File: account_payment_order/models/account_payment_order.py [hunk 1]
```diff
@@ -461,39 +461,51 @@ def _prepare_move_line_partner_account(self, bank_line):
...
+        for bline in blines:
+            total_company_currency += bline.amount_company_currency
+            total_payment_currency += bline.amount_currency
+            partner_ml_vals = self._prepare_move_line_partner_account(
+                bline)
+            mvals['line_ids'].append((0, 0, partner_ml_vals))
+        trf_ml_vals = self._prepare_move_line_offsetting_account(
+            total_company_currency, total_payment_currency, blines)
+        mvals['line_ids'].append((0, 0, trf_ml_vals))
+        move = am_obj.create(mvals)
```
Comment (pedrobaeza) [inline account_payment_order/models/account_payment_order.py:478]: You should add also a prepare method for this dictionary for proper hook design.

Comment (ThomasBinsfeld) [reply account_payment_order/models/account_payment_order.py:478]: I moved all the move dictionary preparation in the `_prepare_move` function.

**Review by pedrobaeza — APPROVED** [review]

**Review by chrisdec — APPROVED** [review]

**Pull request merged**